src/main/scala/cromwell/util/TryUtil.scala

```diff
@@ -76,10 +76,29 @@ object TryUtil extends LazyLogging {
     }
   }
 
-  def sequence[A](s: Seq[Try[A]]): Try[Seq[A]] = {
-    s.collect({case f: Failure[_] => f}) match {
-      case failures if failures.nonEmpty => Failure(new AggregatedException(failures))
-      case _ => Success(s.map(_.get))
+  private def sequenceIterable[T](tries: Iterable[Try[_]], unbox: () => T, prefixErrorMessage: String) = {
+    tries collect { case f: Failure[_] => f } match {
+      case failures if failures.nonEmpty => Failure(new AggregatedException(failures.toSeq, prefixErrorMessage))
+      case _ => Success(unbox())
     }
   }
+
+  def sequence[T](tries: Seq[Try[T]], prefixErrorMessage: String = ""): Try[Seq[T]] = {
+    def unbox = tries map { _.get }
+    sequenceIterable(tries, unbox _, prefixErrorMessage)
+  }
+
+  def sequenceMap[T, U](tries: Map[T, Try[U]], prefixErrorMessage: String = ""): Try[Map[T, U]] = {
+    def unbox = tries mapValues { _.get }
+    sequenceIterable(tries.values, unbox _, prefixErrorMessage)
+  }
+
+  def sequenceMapOrFail[T, U](tries: Map[T, Try[U]], prefixErrorMessage: String = ""): Map[T, U] = {
+    sequenceMap(tries, prefixErrorMessage).get
+  }
+
+  def sequenceOrFail[T](tries: Seq[Try[T]], prefixErrorMessage: String = "") = {
+    sequence(tries, prefixErrorMessage).get
+  }
```

Comment:
They're just `get`s yes. It's just "sugar" and also avoids having to justify the use of `get` as per our code standards, because it makes it clear that you want this to throw an exception if it can't flatten up the Seq/Map.